Makefile

```diff
@@ -196,50 +181,71 @@ talos-azure:
 		convert \
 		-f raw \
 		-o subformat=fixed,force_size \
-		-O vpc /out/disk.raw /out/talos-azure.vhd
-	@tar -C $(PWD)/build -czf $(PWD)/build/$@.tar.gz talos-azure.vhd
-	@rm -rf $(PWD)/build/disk.raw $(PWD)/build/talos-azure.vhd
+		-O vpc /out/azure.raw /out/azure.vhd
+	@tar -C $(PWD)/build -czf $(PWD)/build/azure.tar.gz azure.vhd
+	@rm -rf $(PWD)/build/azure.raw $(PWD)/build/azure.vhd
+ifeq ($(PUSH),"true")
+        ./hack/test/azure-setup.sh
+endif
 
-.PHONY: talos-raw
-talos-raw:
+.PHONY: push-image-azure
+push-image-azure:
+	./hack/test/azure-setup.sh
+
+.PHONY: image-gce
+image-gce:
+	@docker run --rm -v /dev:/dev -v $(PWD)/build:/out --privileged $(DOCKER_ARGS) autonomy/installer:$(TAG) install -n disk -r -p googlecloud -u none
+	@tar -C $(PWD)/build -czf $(PWD)/build/gce.tar.gz disk.raw
+	@rm -rf $(PWD)/build/disk.raw
+ifeq ($(PUSH),"true")
+        ./hack/test/gce-setup.sh
+endif
+
+.PHONY: push-image-gce
+push-image-gce:
+	./hack/test/gce-setup.sh
+
+.PHONY: image-raw
+image-raw:
 	@docker run --rm -v /dev:/dev -v $(PWD)/build:/out --privileged $(DOCKER_ARGS) autonomy/installer:$(TAG) install -n rootfs -r -b
 
-.PHONY: talos
-talos: buildkitd
+.PHONY: iso
+iso:
+	@docker run --rm -i -v $(PWD)/build:/out autonomy/installer:$(TAG) iso
+
+.PHONY: container
+container: buildkitd
 	@$(BINDIR)/buildctl --addr $(BUILDKIT_HOST) \
 		build \
-		--output type=docker,dest=build/$@.tar,name=docker.io/autonomy/$@:$(TAG) \
+		--output type=docker,dest=build/$@.tar,name=docker.io/autonomy/talos:$(TAG) \
 		--opt target=$@ \
 		$(COMMON_ARGS)
 	@docker load < build/$@.tar
 
-.PHONY: osctl-cluster-create
-osctl-cluster-create:
-	@TAG=$(TAG) ./hack/test/$@.sh
-
 .PHONY: basic-integration
 basic-integration:
 	@TAG=$(TAG) ./hack/test/$@.sh
 
-.PHONY: capi-setup
-capi-setup:
+.PHONY: capi
+capi:
 	@TAG=$(TAG) ./hack/test/$@.sh
 
-.PHONY: e2e
+.PHONY: e2e-integration
 e2e-integration:
 	@TAG=$(TAG) ./hack/test/$@.sh
 
-.PHONY: test
-test: buildkitd
+.PHONY: unit-tests
+unit-tests: buildkitd
+	@mkdir -p build
```

Comment:
doing `mkdir build` for unit tests seems unnecessary?

Reply:
Not sure why that is there TBH. Its nothing new, just a rename of `test` to `unit-test`. We can probably remove it.